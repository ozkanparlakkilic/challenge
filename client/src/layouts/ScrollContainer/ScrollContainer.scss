@import '@/styles/scss/_variables';
@import '@/shared/_scroll';

.scroll {
    overflow-y: auto;
    height: $fullHeight;
    width: $fullWidth;
    @include scroll;
}