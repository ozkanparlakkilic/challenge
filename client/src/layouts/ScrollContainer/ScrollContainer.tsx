import React from 'react';
import './ScrollContainer.css';

interface ScrollContainerProps {
    children: React.ReactNode;
    classname?: string;
}

const ScrollContainer = ({ children, classname }: ScrollContainerProps) => {
    return <div className={`${classname ? classname : ''} scroll`}>{children}</div>;
};

export default ScrollContainer;
