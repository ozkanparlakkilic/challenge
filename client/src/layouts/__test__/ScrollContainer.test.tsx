import React from 'react';
import { render } from '@testing-library/react';
import ScrollContainer from '../ScrollContainer/ScrollContainer';

describe('Renderings', () => {
    it('should render the child component', () => {
      const childElement = ("<div>Testing</div>")
      const { container } = render(<ScrollContainer children={childElement} />)
      expect(container).toMatchSnapshot();
    });
});