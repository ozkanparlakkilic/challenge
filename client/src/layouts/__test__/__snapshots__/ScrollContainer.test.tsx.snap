// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderings should render the child component 1`] = `
<div>
  <div
    class=" scroll"
  >
    &lt;div&gt;Testing&lt;/div&gt;
  </div>
</div>
`;
